### PR TITLE
Create the feed endpoint

### PR DESCRIPTION
**Pull request by romiparada**
https://trello.com/c/ZWdGg7Y5

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -12,8 +12,11 @@
 
     resources :tweets, only: %i[create show] do
       post :like, on: :member
+      get :feed, on: :collection
     end
 
+    get 'feed', action: :feed, controller: 'tweets'
```
Comment (micaelalorenzo) [inline config/routes.rb:18]: I think this is duplicated.

File: app/controllers/api/tweets_controller.rb [hunk 2]
```diff
@@ -16,6 +16,14 @@ def like
       render status: :no_content
     end
 
+    def feed
+      feed = Tweet.authored_by(current_user.following_users)
+                  .or(Tweet.liked_by(current_user.following_users)
+                            .where.not(user: current_user))
+                  .order(created_at: :desc)
```
Comment (micaelalorenzo) [inline app/controllers/api/tweets_controller.rb:23]: Could this lead to duplicated tweets? For example, if two users being followed by current_user liked the same tweet.

Comment (romiparada) [reply app/controllers/api/tweets_controller.rb:23]: I tried the following in console:
```
User.find(1) => #<User id:1 ...
User.find(2) => #<User id:2 ...
User.where(id: [1,2,1]).count # => 2
```
So I think given an array of repeated ids the `where` operator returns only one object for each unique id

File: spec/requests/api/tweets/feed_spec.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  subject { get feed_tweets_path, headers:, as: :json }
+
+  let(:user) { create(:user) }
+  let(:user_tweet) { create(:tweet, user:) }
+  let(:headers) { auth_headers(user) }
+  let(:following1) { create(:user) }
+  let!(:tweet1) { create(:tweet, user: following1, created_at: 1.day.ago) }
+  let(:following2) { create(:user) }
+  let!(:tweet2) { create(:like, user: following2).tweet }
+
```
Comment (juan-apa) [inline spec/requests/api/tweets/feed_spec.rb:15]: add some more mock data for when a follower likes a tweet, and expect that one to be displayed

**Review by juan-apa — APPROVED** [review]

**Pull request merged**